### ISSUE TITLE
Among emnist dataset splits, there are cases where labels do not match or out of list occurs.

### ISSUE THREAD

**Issue by kh-mo** [pytorch/vision#2630]
## 🐛 Bug

In the same code,

- 'byclass', 'bymerge', 'balanced' split do not match label & img.
- 'letters' split raises an out of list error.
- 'digits', 'mnist' split work well.

## To Reproduce
```
def imshow(img):
    npimg = img.numpy()
    plt.imshow(np.transpose(npimg, (1, 2, 0)))
    plt.show()
    
def show_img_with_gt(i):
    batch_size=16
    dset_tr = torchvision.datasets.EMNIST(root="./data", split=i, download=True, train=True, 
                                      transform=transforms.Compose([lambda img: torchvision.transforms.functional.rotate(img, -90),
                                                                    transforms.RandomHorizontalFlip(p=1),
                                                                    transforms.ToTensor()]))
    dset_loader = torch.utils.data.DataLoader(dset_tr, batch_size=batch_size)
    i, (image, label) = next(enumerate(dset_loader))
    imshow(torchvision.utils.make_grid(image))
    print('GroundTruth: ', ' '.join('%5s' % dset_tr.classes[label[j]] for j in range(batch_size)))

show_img_with_gt("byclass")
show_img_with_gt("bymerge")
show_img_with_gt("balanced")
show_img_with_gt("letters")
show_img_with_gt("digits")
show_img_with_gt("mnist")
```
![image](https://user-images.githubusercontent.com/32418835/91621794-5b733380-e9cf-11ea-8344-0a989e0af0cc.png)
---------------
![image](https://user-images.githubusercontent.com/32418835/91621805-64fc9b80-e9cf-11ea-8e50-45f89efe1abc.png)
---------------
![image](https://user-images.githubusercontent.com/32418835/91621813-6af27c80-e9cf-11ea-8334-e7f1d9c4c28e.png)
---------------
![image](https://user-images.githubusercontent.com/32418835/91621823-7645a800-e9cf-11ea-95ea-004ee9ca1092.png)
---------------
![image](https://user-images.githubusercontent.com/32418835/91622077-72feec00-e9d0-11ea-8631-a76f91416f86.png)
---------------
![image](https://user-images.githubusercontent.com/32418835/91622088-7befbd80-e9d0-11ea-92fd-f511a0cb41b5.png)

cc @pmeier

**Comment by pmeier** [comment]
HI @kh-mo thanks for the report. I'll look into it, but it might take a while.

**Comment by pmeier** [comment]
I'm pretty covered right now. Any help on this is appreciated.
